### PR TITLE
revamp the test to be more integration specific

### PR DESCRIPTION
**Pull request by ssteveli**

### CODE REVIEW

**Review by brianmaresca — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -2,25 +2,50 @@
...
+GET http://api.qa.fitpay.ninja/compositeUsers/:userId?creditCardState=ERROR&deviceState=FAILED_INITILIZATION
+```
 
 ## API Integration Details
 
 ### Notes
-* The endpoints below will present self-signed SSL certificates, you'll need to either disable SSL validation all together or download and import the certificate into your trusted store.
+* The endpoint will present self-signed SSL certificates, you'll need to either disable SSL validation all together or download and import the certificate into your trusted store.
 * You'll be using our development/testing environment, which at times can be unstable.  You can validate the health of the environment [here](https://api.qa.fitpay.ninja/health).
+* Review the JSON returned by the API to see if there our hints and useful things that can assist in making further API invocations
```
Comment (CarolBloch) [inline README.md]: our --> are

File: README.md [hunk 2]
```diff
@@ -2,25 +2,50 @@
...
 
-**Hint**: How much work can the API do for you?
+* Given the `userId` as part of the URL path, retrieve the underlying API resources from the FitPay API
+* The `Content-Type` returned MUST be `application/json`
+* The JSON returned MUST at least include the following elements
+** `userId`
+** `creditCardId`
+** `state` for both devices and creditCards
+** `deviceId`
+* The `devices` and `credtiCards` MUST support lists of 0-n values
```
Comment (brianmaresca) [inline README.md]: typo `credtiCards`

**Review by CarolBloch — APPROVED** [review]
I like it.  I think it provides enough info to get them going.  One minor comment below.

File: README.md [hunk 3]
```diff
@@ -2,25 +2,50 @@
...
+* Technology choice is whatever platform you're comfortable in
+⋅⋅* If you chose a java based solution, a very basic spring-boot application is included.
 * API credentials and/or bearer tokens can be hard coded with values obtained out-of-band
 * You're familiar with [curl](https://curl.haxx.se) for testing Restful APIs
+* Your solution can ideally be launched with a single command, if not instructions MUST be provided so we can test your new API
+⋅⋅* Example `mvn spring-boot:run`
+
+### Overview
+
+API clients typically need three things when presenting an experience to an end consumer.  The `user`, `devices`, and `creditCards` API resources provided by the FitPay API.  This requires 3 separate network calls to obtain.  The requirement is to create a new API resource called `compositeUser` that allows for all these resources to be returning in a single network call.
```
Comment (Clark73) [inline README.md]: isn't it users not user?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.0.5.RELEASE</version>
+    </parent>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
```
Comment (Clark73) [inline pom.xml]: Should we add the spring boot starter test library? Maybe as a hint?

File: README.md [hunk 5]
```diff
@@ -2,25 +2,50 @@
...
+API resource created that will decrease the number of network calls being made.
+
+### Technology Assumptions
+* Technology choice is whatever platform you're comfortable in
+⋅⋅* If you chose a java based solution, a very basic spring-boot application is included.
 * API credentials and/or bearer tokens can be hard coded with values obtained out-of-band
 * You're familiar with [curl](https://curl.haxx.se) for testing Restful APIs
+* Your solution can ideally be launched with a single command, if not instructions MUST be provided so we can test your new API
+⋅⋅* Example `mvn spring-boot:run`
+
```
Comment (Clark73) [inline README.md]: Should we add a not about submitting the test? Like uploading to a git solution is preferable?

File: README.md (hunk 3 above)
Comment (ssteveli) [reply README.md]: "Technically" no, we're asking for a single API resource which is a single composite user and not for a collection of users.  Subtle.

**Comment by Clark73** [comment]
Looks nice, the only other thing I can think of that we talked about was the "bonus" objectives. Make a UI / Decrypt the User info etc etc

**Review by Clark73 — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (ssteveli) [reply README.md]: I actually don't like things in git, we took that approach in the past and it exposes previous solutions to candidates to "borrow" from.   Not necessarily a good thing for a test.

File: pom.xml (hunk 4 above)
Comment (ssteveli) [reply pom.xml]: The QA in you coming out!

File: README.md (hunk 5 above)
Comment (Clark73) [reply README.md]: Ahh makes sense

**Review by Clark73 — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (Clark73) [reply README.md]: I mean our api resrouce it is not /user it is /users 

https://api.qa.fitpay.ninja/users/:userId

**Comment by Polar-Beard** [comment]
This is a great improvement! I think this will be much more helpful in getting a better idea of the applicant's technical aptitude.

**Review by ciphercules — APPROVED** [review]

**Pull request merged**